zenodo/base/views.py

```diff
@@ -352,6 +355,30 @@ def schemaorg_type(recid=None, bfo=None):
     return 'http://schema.org/CreativeWork'
 
 
+@blueprint.route("/latestdoi/<int:user_id>/<path:repository>", methods=["GET"])
+@ssl_required
+def latest_doi(user_id, repository):
+    """Redirect to the newest record version."""
+    DOI_URL = "http://dx.doi.org/"
+
+    account = get_account(user_id=user_id)
+    if repository not in account.extra_data["repos"]:
+        return abort(404)
+
+    # Get the latest deposition
+    try:
+        dep = account.extra_data["repos"][repository]['depositions'][-1]
+    except IndexError:
+        return abort(404)
+
+    # Extract DOI
+    if "doi" not in dep:
+        return abort(404)
+    doi = dep["doi"]
```

Comment:
You can move this up to the previous try/except block and catch the KeyError to save a couple of lines.